.github/workflows/speech-test-data-ci.yml

```diff
@@ -108,28 +106,29 @@ jobs:
       # data, or by pushing a BASELINE** tag.
       - name: Set environment variables
         run: |
-          echo "::set-env name=TEST_AUDIO_ZIP_FILE::test-audio.zip"
-          echo "::set-env name=TEST_BUILD_FOLDER_PATH::build-speech-test"
+          echo "TEST_AUDIO_ZIP_FILE=test-audio.zip" >> $GITHUB_ENV
+          echo "TEST_BUILD_FOLDER_PATH=build-speech-test" >> $GITHUB_ENV
           if [[ ${GITHUB_REF/refs\/tags\//} == BASELINE* ]]
           then
             echo WORKFLOW TRIGGERED BY A BASELINE TAG.
-            echo ::set-env name=EVENT_ID::${GITHUB_REF/refs\/tags\//}
-            echo "::set-env name=HYPHEN_EVENT_NAME::baseline-tag"
-            echo "::set-env name=IS_BASELINE_TEST::true"
-            echo "::set-env name=UNDERSCORE_EVENT_NAME::baseline_tag"
+            eid=${GITHUB_REF/refs\/tags\//}
+            echo "EVENT_ID=${eid}" >> $GITHUB_ENV
+            echo "HYPHEN_EVENT_NAME=baseline-tag" >> $GITHUB_ENV
+            echo "IS_BASELINE_TEST=true" >> $GITHUB_ENV
+            echo "UNDERSCORE_EVENT_NAME=baseline_tag" >> $GITHUB_ENV
           else
             echo WORKFLOW TRIGGERED BY A TEST DATA UPDATE.
-            echo "::set-env name=EVENT_ID::$(git rev-parse --short HEAD)"
-            echo "::set-env name=HYPHEN_EVENT_NAME::test-data-update"
-            echo "::set-env name=UNDERSCORE_EVENT_NAME::test_data_update"
+            echo "EVENT_ID=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
+            echo "HYPHEN_EVENT_NAME=test-data-update" >> $GITHUB_ENV
+            echo "UNDERSCORE_EVENT_NAME=test_data_update" >> $GITHUB_ENV
           fi
-
-      # https://github.com/msimecek/Azure-Speech-CLI
-      - name: Install and configure Azure Speech CLI
+      # https://github.com/Azure-Samples/cognitive-services-speech-tools
+      - name: Install and configure SPX
         run: |
-          dotnet tool install -g azurespeechcli --version 1.5.2
-          speech config set -n ${{ secrets.SPEECH_PROJECT_NAME }} -k ${{ secrets.SPEECH_SUBSCRIPTION_KEY }} -r ${{ secrets.SPEECH_RESOURCE_REGION }} -s
-
+          dotnet tool install -g spx
+          spx config @name --set ${{ secrets.SPEECH_PROJECT_NAME }} 
+          spx config @key --set ${{ secrets.SPEECH_SUBSCRIPTION_KEY }} 
+          spx config @region --set ${{ secrets.SPEECH_RESOURCE_REGION }}
```

Comment:
I've seen issues where this would log secrets to SPX logs and screen output.
I think Rob added a new switch to make sure these get oviscaped in screen outputs.
If so, you may want to make sure that switch is added here.

Reply:
I'm actually not seeing a screen output issue...My example run is [here](https://github.com/zeigotaro/test-cicd-speech/runs/1445038241?check_suite_focus=true), and the output for setting the key (expanding "Install and configure SPX") is:
```
SPX - Azure Speech CLI, Version 1.0
Copyright (c) 2020 Microsoft Corporation. All Rights Reserved.

key (saved at '/home/runner/work/test-cicd-speech/test-cicd-speech')

  ***
```

Reply:
I remember some mention in the GitHub Actions docs that this was a risk in some situations, but @glharper like you're saying we never actually ran into that issue. I'll try and track that doc down